include/es/pooled_resource.h

```diff
@@ -82,8 +84,12 @@ class managed_resource_pool {
         if(d_map.find(idx) == d_map.end()){
             boost::mutex::scoped_lock(_map_lock);
             d_map[idx] = boost::shared_ptr< pooled_resource <T > >(new pooled_resource<T>(d_max) );
+            while(d_map[idx].get() == 0)
```

Comment:
would be nice to comment on this -- looks like we are busy looping until we can acquire one to populate --